server/ruby/app.rb

```diff
@@ -0,0 +1,215 @@
+require 'stripe'
+require 'sinatra'
+require 'sinatra/cookies'
+# require 'sinatra/reloader'
+require 'dotenv'
+require 'json'
+require_relative 'inventory'
+require_relative 'setup'
+
+Dotenv.load(File.dirname(__FILE__) + '/../../.env')
+Stripe.api_key = ENV['STRIPE_SECRET_KEY']
+
+set :static, true
+set :root, File.dirname(__FILE__)
+set :public_folder, Dir.chdir(Dir.pwd + '/../../public')
+
+
+get '/javascripts/:path' do
+  content_type 'text/javascript'
+  send_file "javascripts/#{params['path']}"
+end
+
+get '/stylesheets/:path' do
+  content_type 'text/css'
+  send_file "stylesheets/#{params['path']}"
+end
+
+get '/images/*.*' do |path, ext|
+  if ext == "svg"
+    content_type "image/#{ext}+xml"
+  else
+    content_type "image/#{ext}"
+  end
+  send_file "images/#{path}.#{ext}"
+end
+
+get '/' do
+  # Route to the index page which will show our cart 
+  content_type 'text/html'
+  send_file 'index.html'
+end
+
+get '/config' do
+  # Route to return configurations details required by the frontend
+  content_type 'application/json'
+  {
+    'stripePublishableKey': ENV['STRIPE_PUBLISHABLE_KEY'],
+    'stripeCountry': ENV['STRIPE_ACCOUNT_COUNTRY'] || 'US',
+    'country': 'US',
+    'currency': 'eur',
+    'paymentMethods': ENV['PAYMENT_METHODS'] ? ENV['PAYMENT_METHODS'].split(', ') : ['card'],
+    'shippingOptions': [
+      {
+        'id': 'free',
+        'label': 'Free Shipping',
+      'detail': 'Delivery within 5 days',
+        'amount': 0,
+      },
+      {
+        'id': 'express',
+          'label': 'Express Shipping',
+          'detail': 'Next day delivery',
+          'amount': 500,
+      }
+    ]
+  }.to_json
+end
+
+get '/products' do
+  content_type 'application/json'
+  products = Inventory.list_products
+  if Inventory.products_exist(products)
+    products.to_json
+  else
+    # Setup products
+    puts "Needs to setup products"
+    create_data
+    products = Inventory.list_products
+    products.to_json
+  end
+end
+
+get '/products/:product_id' do
+  content_type 'application/json'
+  product = Inventory.retrieve_product(params['product_id'])
+  product.to_json
+end
+
+post '/payment_intents' do
+  content_type 'application/json'
+  data = JSON.parse request.body.read
+
+  payment_intent = Stripe::PaymentIntent.create(
+    amount: Inventory.calculate_payment_amount(data['items']),
+    currency: data['currency'],
+    payment_method_types: ENV['PAYMENT_METHODS'] ? ENV['PAYMENT_METHODS'].split(', ') : ['card']
+  )
+
+  {
+    paymentIntent: payment_intent
+  }.to_json
+end
+
+post '/payment_intents/:id/shipping_change' do
+  content_type 'application/json'
+  data = JSON.parse request.body.read
+
+  amount = Inventory.calculate_payment_amount(data['items'])
+  amount += Inventory.get_shipping_cost(data['shippingOption']['id'])
```

Comment:
Throws error `nil can't be coerced into Integer`
![image](https://user-images.githubusercontent.com/23213994/55346056-cc987980-54b1-11e9-8cdb-29736b434d63.png)

Reply:
Fixed in https://github.com/stripe/stripe-payments-demo/pull/54/commits/936a1f7c762bc851c33f217852850d49ca7d4915